thoth-api/migrations/0.9.0/up.sql

```diff
@@ -0,0 +1,37 @@
+CREATE TABLE reference (
+    reference_id            UUID PRIMARY KEY DEFAULT uuid_generate_v4(),
+    work_id                 UUID NOT NULL REFERENCES work(work_id) ON DELETE CASCADE,
+    reference_ordinal       INTEGER NOT NULL CHECK (reference_ordinal > 0),
+    doi                     TEXT CHECK (doi ~ '^https:\/\/doi\.org\/10\.\d{4,9}\/[-._\;\(\)\/:a-zA-Z0-9]+$'),
```

Comment:
This regex is very slightly different from the one for `work` `doi` in 0.1.0 (e.g. `~*` vs `~`) - ideally they should match.